tree/treeplayer/src/TTreeFormula.cxx

```diff
@@ -5373,7 +5373,7 @@ bool TTreeFormula::LoadCurrentDim() {
       // variable (expected for the first one) can NOT be done via negative values of
       // fCumulSizes.
 
-      TLeaf *leaf = (TLeaf*)fLeaves.UncheckedAt(i);
+      TLeaf *leaf = i <= fLeaves.GetLast() ? (TLeaf *)fLeaves.UncheckedAt(i) : nullptr;
```

Comment:
```suggestion
      auto *leaf = static_cast<TLeaf *>(fLeaves.At(i));
```

Reply:
I don't think this is equivalent, as it will print a message and raise an "Error". Is the change of behavior intended?

Reply:
I am actually not sure. On the one hand, your proposal indeed does not "interfere too much", silently initialising the variable to `nullptr`. On the other hand, if I read the following lines, especially line 5390 where the `leaf` variable is accessed without a safety check, I am inclined to say we should be very explicit that something went wrong and probably printing the error message is not even enough, we should abort the function (i.e. raise an exception). But to be honest that's just my opinion and we probably also want @pcanal's.

Reply:
Although raising an exception here would prevent the code of lines 5377-5386 to run, so I am again undecided on what is the best course of action

Reply:
I do not recall the case covered by out of bound index, however stylistically the code current in the PR matches other 'get the leaf pointer' code snippet. (See the commits mentioned earlier).  The execution of line 5383 and 5384 in the error case are necessary to leaf the `TTreeFormula` is a usable state.